<a href="https://colab.research.google.com/github/alexlopesdesouzagearbest-debug/TrabalhoR/blob/main/EstatisticaBasica_Python_Aula01_Solucao_Aula01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Data Analysis & Predictive Insights


## Turma 92BI

### Trabalho prático usando o Python

### Habilidades desenvolvidas: análise descritiva dos dados.


![dataanalytics.png](attachment:dataanalytics.png)

![dadosbrutos.png](attachment:dadosbrutos.png)

## Ferramenta

![scipyorg.png](attachment:scipyorg.png)

### Instalar as bibliotecas pandas, numpy, matplotlib e seaborn

In [1]:
# Importar as bibliotecas
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# opcao de mudar a notacao cientifica para aritmerica
pd.set_option('display.float_format','{:.2f}'.format)

In [3]:
# Importar a base de dados
cadastro = pd.read_csv('cadastro.csv')

In [4]:
# Visualizar os dados
cadastro.head()

Unnamed: 0      NUM_CPF  CHAVE_CONTRATO DATA_NASCIMENTO  RENDA_PRESUMIDA  \
0           1  40977545806               1      25/09/1996       1890000.00   
1           2  72544350725               2      10/03/1961       1333890.24   
2           3  10536099812               3      28/11/1965         22000.00   
3           4   1471742873               4      02/05/1959         16000.00   
4           5    954396332               5      10/11/1984         15929.00   

       CEP                  LOGRADOURO  NUMERO COMPLEMENTO    CEP_A  ...  \
0  2471020  R AUGUSTO VIANA DE CASTELO   26.00         NaN  2471020  ...   
1  1319001               R MARIA PAULA  201.00       AP 52  1319001  ...   
2  4323120                 R DAS JOIAS  202.00       VL DO  4323120  ...   
3  4136030                 R GUARAREMA  430.00         NaN  4136030  ...   
4  8452105   R JOAO DE SIQUEIRA AFONSO  709.00         CAS  8452105  ...   

  NUM_AGENCIA  NUM_CONTA MARCA_VEICULO  MODELO_VEICULO  PLACA_VEICULO  \
0         NaN        NaN           NaN             NaN            NaN   
1     6571.00   15044.00       FERRARI      355 SPIDER        AFZ6118   
2     7006.00   52946.00           NaN             NaN            NaN   
3     1850.00 1103486.00           NaN             NaN            NaN   
4      123.00  580882.00           NaN             NaN            NaN   

        CNPJ_CREDOR  VALOR_DIVIDA  STATUS_CONSENTIMENTO       DATA_INCLUSAO1  \
0  3774819001508.00           220                     1   2/26/2019 11:28:49   
1 40432544000147.00           850                     1     2/7/2001 0:00:00   
2 61695227000193.00           142                     1    8/23/2018 6:42:12   
3               NaN             0                     1                  NaN   
4 35109651000107.00           400                     1  11/24/2010 10:30:45   

   IDADE  
0  23.00  
1  40.00  
2  53.00  
3    NaN  
4  26.00  

[5 rows x 30 columns]

## Tipos de variáveis
![escalamensuracao.png](attachment:escalamensuracao.png)

In [5]:
# Tipos de variáveis
cadastro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 426 entries, 0 to 425
Data columns (total 30 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            426 non-null    int64  
 1   NUM_CPF               426 non-null    int64  
 2   CHAVE_CONTRATO        426 non-null    int64  
 3   DATA_NASCIMENTO       426 non-null    object 
 4   RENDA_PRESUMIDA       426 non-null    float64
 5   CEP                   426 non-null    int64  
 6   LOGRADOURO            426 non-null    object 
 7   NUMERO                425 non-null    float64
 8   COMPLEMENTO           206 non-null    object 
 9   CEP_A                 426 non-null    int64  
 10  BAIRRO                390 non-null    object 
 11  CIDADE                426 non-null    object 
 12  UF                    426 non-null    object 
 13  LATITUDE              389 non-null    float64
 14  LONGITUDE             389 non-null    float64
 15  DDD_CELULAR           2

In [6]:
# Contagem dos dados
cadastro.count()

Unnamed: 0              426
NUM_CPF                 426
CHAVE_CONTRATO          426
DATA_NASCIMENTO         426
RENDA_PRESUMIDA         426
CEP                     426
LOGRADOURO              426
NUMERO                  425
COMPLEMENTO             206
CEP_A                   426
BAIRRO                  390
CIDADE                  426
UF                      426
LATITUDE                389
LONGITUDE               389
DDD_CELULAR             275
CELULAR                 275
DDD_CELULAR_2           170
CELULAR_2               170
COD_BANCO               308
NUM_AGENCIA             308
NUM_CONTA               308
MARCA_VEICULO             2
MODELO_VEICULO            2
PLACA_VEICULO             2
CNPJ_CREDOR             421
VALOR_DIVIDA            426
STATUS_CONSENTIMENTO    426
DATA_INCLUSAO1          421
IDADE                   421
dtype: int64

In [7]:
# Contagem de valores únicos
cadastro.nunique()

Unnamed: 0              426
NUM_CPF                 311
CHAVE_CONTRATO          426
DATA_NASCIMENTO         305
RENDA_PRESUMIDA         317
CEP                     310
LOGRADOURO              308
NUMERO                  241
COMPLEMENTO              86
CEP_A                   310
BAIRRO                  230
CIDADE                    1
UF                        1
LATITUDE                282
LONGITUDE               282
DDD_CELULAR              18
CELULAR                 202
DDD_CELULAR_2            14
CELULAR_2               124
COD_BANCO                17
NUM_AGENCIA             294
NUM_CONTA               308
MARCA_VEICULO             2
MODELO_VEICULO            2
PLACA_VEICULO             2
CNPJ_CREDOR             153
VALOR_DIVIDA            235
STATUS_CONSENTIMENTO      1
DATA_INCLUSAO1          405
IDADE                    54
dtype: int64

In [8]:
# Retorna verdadeiro ou falso para presença de dados faltantes (missing value)
cadastro.isna()

Unnamed: 0  NUM_CPF  CHAVE_CONTRATO  DATA_NASCIMENTO  RENDA_PRESUMIDA  \
0         False    False           False            False            False   
1         False    False           False            False            False   
2         False    False           False            False            False   
3         False    False           False            False            False   
4         False    False           False            False            False   
..          ...      ...             ...              ...              ...   
421       False    False           False            False            False   
422       False    False           False            False            False   
423       False    False           False            False            False   
424       False    False           False            False            False   
425       False    False           False            False            False   

       CEP  LOGRADOURO  NUMERO  COMPLEMENTO  CEP_A  ...  NUM_AGENCIA  \
0    False       False   False         True  False  ...         True   
1    False       False   False        False  False  ...        False   
2    False       False   False        False  False  ...        False   
3    False       False   False         True  False  ...        False   
4    False       False   False        False  False  ...        False   
..     ...         ...     ...          ...    ...  ...          ...   
421  False       False   False         True  False  ...        False   
422  False       False   False        False  False  ...         True   
423  False       False   False         True  False  ...         True   
424  False       False   False        False  False  ...        False   
425  False       False   False        False  False  ...        False   

     NUM_CONTA  MARCA_VEICULO  MODELO_VEICULO  PLACA_VEICULO  CNPJ_CREDOR  \
0         True           True            True           True        False   
1        False          False           False          False        False   
2        False           True            True           True        False   
3        False           True            True           True         True   
4        False           True            True           True        False   
..         ...            ...             ...            ...          ...   
421      False           True            True           True        False   
422       True           True            True           True        False   
423       True           True            True           True        False   
424      False           True            True           True        False   
425      False           True            True           True        False   

     VALOR_DIVIDA  STATUS_CONSENTIMENTO  DATA_INCLUSAO1  IDADE  
0           False                 False           False  False  
1           False                 False           False  False  
2           False                 False           False  False  
3           False                 False            True   True  
4           False                 False           False  False  
..            ...                   ...             ...    ...  
421         False                 False           False  False  
422         False                 False           False  False  
423         False                 False           False  False  
424         False                 False           False  False  
425         False                 False           False  False  

[426 rows x 30 columns]

In [9]:
# Soma o número de dados faltantes (missing value)
cadastro.isna().sum()

Unnamed: 0                0
NUM_CPF                   0
CHAVE_CONTRATO            0
DATA_NASCIMENTO           0
RENDA_PRESUMIDA           0
CEP                       0
LOGRADOURO                0
NUMERO                    1
COMPLEMENTO             220
CEP_A                     0
BAIRRO                   36
CIDADE                    0
UF                        0
LATITUDE                 37
LONGITUDE                37
DDD_CELULAR             151
CELULAR                 151
DDD_CELULAR_2           256
CELULAR_2               256
COD_BANCO               118
NUM_AGENCIA             118
NUM_CONTA               118
MARCA_VEICULO           424
MODELO_VEICULO          424
PLACA_VEICULO           424
CNPJ_CREDOR               5
VALOR_DIVIDA              0
STATUS_CONSENTIMENTO      0
DATA_INCLUSAO1            5
IDADE                     5
dtype: int64

In [10]:
# Calcular o porcentagem de dados faltantes (missing value)
cadastro.isna().sum()/len(cadastro)*100

Unnamed: 0              0.00
NUM_CPF                 0.00
CHAVE_CONTRATO          0.00
DATA_NASCIMENTO         0.00
RENDA_PRESUMIDA         0.00
CEP                     0.00
LOGRADOURO              0.00
NUMERO                  0.23
COMPLEMENTO            51.64
CEP_A                   0.00
BAIRRO                  8.45
CIDADE                  0.00
UF                      0.00
LATITUDE                8.69
LONGITUDE               8.69
DDD_CELULAR            35.45
CELULAR                35.45
DDD_CELULAR_2          60.09
CELULAR_2              60.09
COD_BANCO              27.70
NUM_AGENCIA            27.70
NUM_CONTA              27.70
MARCA_VEICULO          99.53
MODELO_VEICULO         99.53
PLACA_VEICULO          99.53
CNPJ_CREDOR             1.17
VALOR_DIVIDA            0.00
STATUS_CONSENTIMENTO    0.00
DATA_INCLUSAO1          1.17
IDADE                   1.17
dtype: float64

In [11]:
# Análise descritiva das variáveis
cadastro.describe().round(2)


Unnamed: 0        NUM_CPF  CHAVE_CONTRATO  RENDA_PRESUMIDA        CEP  \
count      426.00         426.00          426.00           426.00     426.00   
mean       213.50 26830502524.11          213.50         10387.81 4635212.33   
std        123.12 19135846183.67          123.12        111783.91 1978839.32   
min          1.00    94700508.00            1.00             0.00 1007020.00   
25%        107.25 10869067862.00          107.25          2068.50 3153250.25   
50%        213.50 26041138816.00          213.50          2431.58 4323120.00   
75%        319.75 35798933865.00          319.75          2923.54 5644782.50   
max        426.00 99921022504.00          426.00       1890000.00 8490018.00   

        NUMERO      CEP_A  LATITUDE  LONGITUDE  DDD_CELULAR      CELULAR  \
count   425.00     426.00    389.00     389.00       275.00       275.00   
mean    595.29 4635212.33    -23.58     -46.60        16.50 976736587.11   
std    4869.60 1978839.32      0.08       0.11        17.78  56323647.16   
min       3.00 1007020.00    -23.80     -46.81        11.00  78462120.00   
25%      58.00 3153250.25    -23.63     -46.68        11.00 969555131.50   
50%     164.00 4323120.00    -23.56     -46.63        11.00 982497429.00   
75%     430.00 5644782.50    -23.51     -46.52        11.00 992424488.50   
max   99999.00 8490018.00    -23.43     -46.38        98.00 999993969.00   

       DDD_CELULAR_2    CELULAR_2  COD_BANCO  NUM_AGENCIA     NUM_CONTA  \
count         170.00       170.00     308.00       308.00        308.00   
mean           18.26 960391066.33     189.76      2570.55   51862517.85   
std            19.58 138162785.25     175.78      2347.82  469016468.99   
min            11.00  77151941.00       1.00         1.00        290.00   
25%            11.00 973504991.25      33.00       563.25      15030.00   
50%            11.00 982267694.50     237.00      1900.00      53951.00   
75%            11.00 993495586.00     341.00      3781.00     488489.25   
max            98.00 999889639.00     756.00      9270.00 6500059293.00   

            CNPJ_CREDOR  VALOR_DIVIDA  STATUS_CONSENTIMENTO   IDADE  
count            421.00        426.00                426.00  421.00  
mean  31873758302107.56     212868.56                  1.00   45.74  
std   22283968587101.65    2202077.89                  0.00  107.73  
min     113451000135.00          0.00                  1.00   17.00  
25%    8623036000133.00        198.50                  1.00   30.00  
50%   40432544000147.00        360.00                  1.00   37.00  
75%   43374768000138.00        981.00                  1.00   46.00  
max   96367305000105.00   28072009.00                  1.00 2006.00

In [12]:
# Mudar o formato da saída da análise descritiva
Describe=cadastro.describe().round(2)
print(Describe)

       Unnamed: 0        NUM_CPF  CHAVE_CONTRATO  RENDA_PRESUMIDA        CEP  \
count      426.00         426.00          426.00           426.00     426.00   
mean       213.50 26830502524.11          213.50         10387.81 4635212.33   
std        123.12 19135846183.67          123.12        111783.91 1978839.32   
min          1.00    94700508.00            1.00             0.00 1007020.00   
25%        107.25 10869067862.00          107.25          2068.50 3153250.25   
50%        213.50 26041138816.00          213.50          2431.58 4323120.00   
75%        319.75 35798933865.00          319.75          2923.54 5644782.50   
max        426.00 99921022504.00          426.00       1890000.00 8490018.00   

        NUMERO      CEP_A  LATITUDE  LONGITUDE  DDD_CELULAR      CELULAR  \
count   425.00     426.00    389.00     389.00       275.00       275.00   
mean    595.29 4635212.33    -23.58     -46.60        16.50 976736587.11   
std    4869.60 1978839.32      0.08       0.11     

In [29]:
#Preprocessing
# mudar o formato da variável number para string
cadastro['NUM_CPF'] = cadastro['NUM_CPF'].astype(str)
cadastro['CHAVE_CONTRATO'] = cadastro['CHAVE_CONTRATO'].astype(str)
cadastro['CEP'] = cadastro['CEP'].astype(str)
cadastro['LATITUDE'] = cadastro['LATITUDE'].astype(str)
cadastro['LONGITUDE'] = cadastro['LONGITUDE'].astype(str)
cadastro['DDD_CELULAR'] = cadastro['DDD_CELULAR'].astype(str)
cadastro['DDD_CELULAR_2'] = cadastro['DDD_CELULAR_2'].astype(str)
cadastro['CELULAR_2'] = cadastro['CELULAR_2'].astype(str)
cadastro['CELULAR'] = cadastro['CELULAR'].astype(str)
cadastro['COD_BANCO'] = cadastro['COD_BANCO'].astype(str)
cadastro['NUM_AGENCIA'] = cadastro['NUM_AGENCIA'].astype(str)
cadastro['NUM_CONTA'] = cadastro['NUM_CONTA'].astype(str)
cadastro['CNPJ_CREDOR'] = cadastro['CNPJ_CREDOR'].astype(str)


In [15]:
# Tipos de variáveis
cadastro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 426 entries, 0 to 425
Data columns (total 30 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            426 non-null    int64  
 1   NUM_CPF               426 non-null    object 
 2   CHAVE_CONTRATO        426 non-null    object 
 3   DATA_NASCIMENTO       426 non-null    object 
 4   RENDA_PRESUMIDA       426 non-null    float64
 5   CEP                   426 non-null    int64  
 6   LOGRADOURO            426 non-null    object 
 7   NUMERO                425 non-null    float64
 8   COMPLEMENTO           206 non-null    object 
 9   CEP_A                 426 non-null    int64  
 10  BAIRRO                390 non-null    object 
 11  CIDADE                426 non-null    object 
 12  UF                    426 non-null    object 
 13  LATITUDE              389 non-null    float64
 14  LONGITUDE             389 non-null    float64
 15  DDD_CELULAR           2

In [30]:
# Análise descritiva das variáveis
cadastro.describe().round(2)

Unnamed: 0  RENDA_PRESUMIDA   NUMERO      CEP_A  VALOR_DIVIDA  \
count      426.00           426.00   425.00     426.00        426.00   
mean       213.50         10387.81   595.29 4635212.33     212868.56   
std        123.12        111783.91  4869.60 1978839.32    2202077.89   
min          1.00             0.00     3.00 1007020.00          0.00   
25%        107.25          2068.50    58.00 3153250.25        198.50   
50%        213.50          2431.58   164.00 4323120.00        360.00   
75%        319.75          2923.54   430.00 5644782.50        981.00   
max        426.00       1890000.00 99999.00 8490018.00   28072009.00   

       STATUS_CONSENTIMENTO   IDADE  
count                426.00  421.00  
mean                   1.00   45.74  
std                    0.00  107.73  
min                    1.00   17.00  
25%                    1.00   30.00  
50%                    1.00   37.00  
75%                    1.00   46.00  
max                    1.00 2006.00

In [31]:
# Soma o número de dados faltantes (missing value)
cadastro.isna().sum()

Unnamed: 0                0
NUM_CPF                   0
CHAVE_CONTRATO            0
DATA_NASCIMENTO           0
RENDA_PRESUMIDA           0
CEP                       0
LOGRADOURO                0
NUMERO                    1
COMPLEMENTO             220
CEP_A                     0
BAIRRO                   36
CIDADE                    0
UF                        0
LATITUDE                  0
LONGITUDE                 0
DDD_CELULAR               0
CELULAR                   0
DDD_CELULAR_2             0
CELULAR_2                 0
COD_BANCO                 0
NUM_AGENCIA               0
NUM_CONTA                 0
MARCA_VEICULO           424
MODELO_VEICULO          424
PLACA_VEICULO           424
CNPJ_CREDOR               0
VALOR_DIVIDA              0
STATUS_CONSENTIMENTO      0
DATA_INCLUSAO1            5
IDADE                     5
dtype: int64

In [18]:
# Transpor as duas primeiras linhas da base de dados para visalização
cadastro.head(2).T

0                 1
Unnamed: 0                                     1                 2
NUM_CPF                              40977545806       72544350725
CHAVE_CONTRATO                                 1                 2
DATA_NASCIMENTO                       25/09/1996        10/03/1961
RENDA_PRESUMIDA                       1890000.00        1333890.24
CEP                                      2471020           1319001
LOGRADOURO            R AUGUSTO VIANA DE CASTELO     R MARIA PAULA
NUMERO                                     26.00            201.00
COMPLEMENTO                                  NaN             AP 52
CEP_A                                    2471020           1319001
BAIRRO                             VILA BASILEIA        BELA VISTA
CIDADE                                 SAO PAULO         SAO PAULO
UF                                            SP                SP
LATITUDE                                     NaN               NaN
LONGITUDE                                    NaN               NaN
DDD_CELULAR                                11.00             11.00
CELULAR                             977060299.00      972545525.00
DDD_CELULAR_2                                NaN             11.00
CELULAR_2                                    NaN      985714369.00
COD_BANCO                                    NaN            341.00
NUM_AGENCIA                                  NaN           6571.00
NUM_CONTA                                    NaN          15044.00
MARCA_VEICULO                                NaN           FERRARI
MODELO_VEICULO                               NaN        355 SPIDER
PLACA_VEICULO                                NaN           AFZ6118
CNPJ_CREDOR                     3774819001508.00 40432544000147.00
VALOR_DIVIDA                                 220               850
STATUS_CONSENTIMENTO                           1                 1
DATA_INCLUSAO1                2/26/2019 11:28:49  2/7/2001 0:00:00
IDADE                                      23.00             40.00

In [19]:
# Mudar a categoria "nan" para "NaN" (identificação de missing data)
cadastro.replace('nan', np.nan, inplace=True)

In [20]:
# Transpor as duas primeiras linhas da base de dados para visalização
cadastro.head(2).T

0                 1
Unnamed: 0                                     1                 2
NUM_CPF                              40977545806       72544350725
CHAVE_CONTRATO                                 1                 2
DATA_NASCIMENTO                       25/09/1996        10/03/1961
RENDA_PRESUMIDA                       1890000.00        1333890.24
CEP                                      2471020           1319001
LOGRADOURO            R AUGUSTO VIANA DE CASTELO     R MARIA PAULA
NUMERO                                     26.00            201.00
COMPLEMENTO                                  NaN             AP 52
CEP_A                                    2471020           1319001
BAIRRO                             VILA BASILEIA        BELA VISTA
CIDADE                                 SAO PAULO         SAO PAULO
UF                                            SP                SP
LATITUDE                                     NaN               NaN
LONGITUDE                                    NaN               NaN
DDD_CELULAR                                11.00             11.00
CELULAR                             977060299.00      972545525.00
DDD_CELULAR_2                                NaN             11.00
CELULAR_2                                    NaN      985714369.00
COD_BANCO                                    NaN            341.00
NUM_AGENCIA                                  NaN           6571.00
NUM_CONTA                                    NaN          15044.00
MARCA_VEICULO                                NaN           FERRARI
MODELO_VEICULO                               NaN        355 SPIDER
PLACA_VEICULO                                NaN           AFZ6118
CNPJ_CREDOR                     3774819001508.00 40432544000147.00
VALOR_DIVIDA                                 220               850
STATUS_CONSENTIMENTO                           1                 1
DATA_INCLUSAO1                2/26/2019 11:28:49  2/7/2001 0:00:00
IDADE                                      23.00             40.00

In [21]:
# Soma o número de dados faltantes (missing value)
cadastro.isna().sum()

Unnamed: 0                0
NUM_CPF                   0
CHAVE_CONTRATO            0
DATA_NASCIMENTO           0
RENDA_PRESUMIDA           0
CEP                       0
LOGRADOURO                0
NUMERO                    1
COMPLEMENTO             220
CEP_A                     0
BAIRRO                   36
CIDADE                    0
UF                        0
LATITUDE                 37
LONGITUDE                37
DDD_CELULAR             151
CELULAR                 151
DDD_CELULAR_2           256
CELULAR_2               256
COD_BANCO               118
NUM_AGENCIA             118
NUM_CONTA               118
MARCA_VEICULO           424
MODELO_VEICULO          424
PLACA_VEICULO           424
CNPJ_CREDOR               5
VALOR_DIVIDA              0
STATUS_CONSENTIMENTO      0
DATA_INCLUSAO1            5
IDADE                     5
dtype: int64

In [22]:
# Calcular o porcentagem de dados faltantes (missing value)
cadastro.isna().sum()/len(cadastro)*100

Unnamed: 0              0.00
NUM_CPF                 0.00
CHAVE_CONTRATO          0.00
DATA_NASCIMENTO         0.00
RENDA_PRESUMIDA         0.00
CEP                     0.00
LOGRADOURO              0.00
NUMERO                  0.23
COMPLEMENTO            51.64
CEP_A                   0.00
BAIRRO                  8.45
CIDADE                  0.00
UF                      0.00
LATITUDE                8.69
LONGITUDE               8.69
DDD_CELULAR            35.45
CELULAR                35.45
DDD_CELULAR_2          60.09
CELULAR_2              60.09
COD_BANCO              27.70
NUM_AGENCIA            27.70
NUM_CONTA              27.70
MARCA_VEICULO          99.53
MODELO_VEICULO         99.53
PLACA_VEICULO          99.53
CNPJ_CREDOR             1.17
VALOR_DIVIDA            0.00
STATUS_CONSENTIMENTO    0.00
DATA_INCLUSAO1          1.17
IDADE                   1.17
dtype: float64

## Imputação de dados faltantes

In [32]:
# Imputação de dados faltantes usando a média
cadastro['Idade_i'] = cadastro['IDADE'].fillna(value=cadastro['IDADE'].mean())

In [33]:
# Análise descritiva das variáveis
cadastro.describe().round(2)

Unnamed: 0  RENDA_PRESUMIDA   NUMERO      CEP_A  VALOR_DIVIDA  \
count      426.00           426.00   425.00     426.00        426.00   
mean       213.50         10387.81   595.29 4635212.33     212868.56   
std        123.12        111783.91  4869.60 1978839.32    2202077.89   
min          1.00             0.00     3.00 1007020.00          0.00   
25%        107.25          2068.50    58.00 3153250.25        198.50   
50%        213.50          2431.58   164.00 4323120.00        360.00   
75%        319.75          2923.54   430.00 5644782.50        981.00   
max        426.00       1890000.00 99999.00 8490018.00   28072009.00   

       STATUS_CONSENTIMENTO   IDADE  Idade_i  
count                426.00  421.00   426.00  
mean                   1.00   45.74    45.74  
std                    0.00  107.73   107.09  
min                    1.00   17.00    17.00  
25%                    1.00   30.00    30.00  
50%                    1.00   37.00    37.00  
75%                    1.00   46.00    46.00  
max                    1.00 2006.00  2006.00

### Coeficiente de variação (CV)

In [34]:
# Cálculo do coeficiente de variação (CV)
# CV = desvio padrão/média

desc = cadastro.describe().T

desc['CV'] = desc['std']/desc['mean']

In [35]:
desc.round(2)

count       mean        std        min        25%  \
Unnamed: 0           426.00     213.50     123.12       1.00     107.25   
RENDA_PRESUMIDA      426.00   10387.81  111783.91       0.00    2068.50   
NUMERO               425.00     595.29    4869.60       3.00      58.00   
CEP_A                426.00 4635212.33 1978839.32 1007020.00 3153250.25   
VALOR_DIVIDA         426.00  212868.56 2202077.89       0.00     198.50   
STATUS_CONSENTIMENTO 426.00       1.00       0.00       1.00       1.00   
IDADE                421.00      45.74     107.73      17.00      30.00   
Idade_i              426.00      45.74     107.09      17.00      30.00   

                            50%        75%         max    CV  
Unnamed: 0               213.50     319.75      426.00  0.58  
RENDA_PRESUMIDA         2431.58    2923.54  1890000.00 10.76  
NUMERO                   164.00     430.00    99999.00  8.18  
CEP_A                4323120.00 5644782.50  8490018.00  0.43  
VALOR_DIVIDA             360.00     981.00 28072009.00 10.34  
STATUS_CONSENTIMENTO       1.00       1.00        1.00  0.00  
IDADE                     37.00      46.00     2006.00  2.36  
Idade_i                   37.00      46.00     2006.00  2.34

# Interpretação do Coeficiente de Variação (CV)


![image.png](attachment:75b909ca-d40c-454e-8b33-ce14616a9379.png)


Renda presumida: O desvio padrão é 10.76 vezes o valor da média. Isso indica presença de outliers. Quantos outliers?

Recomendado é até 0,3 ou 30%. Valores acima de 30% indicam presença de outliers. (Critério do IBGE)
    


In [ ]:
# Análise descritiva
print(desc)

### Análise exploratória da Variável de análise: RENDA_PRESUMIDA

In [ ]:
## Análise de detecção de outliers usando o Box Plot
sns.boxplot(y="RENDA_PRESUMIDA", data=cadastro)

In [ ]:
# filtro de renda < 10000
renda = cadastro[(cadastro['RENDA_PRESUMIDA'] <10000)]

In [ ]:
# Box Plot
sns.boxplot(y="RENDA_PRESUMIDA", data=renda)

## Cálculo dos limites que definem outliers e pontos extremos do Box Plot

In [ ]:
# Cálculo do mínimo
print('Minimo = ', min(cadastro['RENDA_PRESUMIDA']))

In [ ]:
# Cálculo do máximo
print('Maximo = ', max(cadastro['RENDA_PRESUMIDA']))

In [ ]:
# Cálculo do quartil 1
Q1 = np.percentile(cadastro['RENDA_PRESUMIDA'], 25)
print(Q1)
print('Quartil1 (Q1) = ', np.percentile(cadastro['RENDA_PRESUMIDA'], 25))

In [ ]:
# Cálculo do quartil 2
Q2=np.percentile(cadastro['RENDA_PRESUMIDA'], 50)
print(Q2)
print('Quartil2 (Q2) = ', np.percentile(cadastro['RENDA_PRESUMIDA'], 50))

In [ ]:
# Cálculo do quartil 3
Q3=np.percentile(cadastro['RENDA_PRESUMIDA'], 75)
print(Q3)
print('Quartil3 (Q3) = ', np.percentile(cadastro['RENDA_PRESUMIDA'], 75))

In [ ]:
# Cálculo do interquartil (IQ)
IQ = np.percentile(cadastro['RENDA_PRESUMIDA'], 75) - np.percentile(cadastro['RENDA_PRESUMIDA'], 25)
print(IQ)
print('Inter-quartil (IQ) =', np.percentile(cadastro['RENDA_PRESUMIDA'], 75) - np.percentile(cadastro['RENDA_PRESUMIDA'], 25))

In [ ]:
# Limite 1
limite1 = Q3+3*IQ
print(limite1)

In [ ]:
#Limite 2
limite2 = Q3+1.5*IQ
print(limite2)

In [ ]:
#Limite 3
limite3=Q1-1.5*IQ
print(limite3)

In [ ]:
#Limite 4
limite4 = Q1-3*IQ
print(limite4)

## Box Plot da Renda Presumida


In [ ]:
# Criar uma variável outlier_renda no Data Frame, sendo 0 = não outlier, 1= outlier e 2 = ponto extremo
cadastro.loc[cadastro.RENDA_PRESUMIDA > limite1 , 'outlier_renda']=2
cadastro.loc[(cadastro.RENDA_PRESUMIDA >= limite2) & (cadastro.RENDA_PRESUMIDA <= limite1), 'outlier_renda']=1
cadastro.loc[(cadastro.RENDA_PRESUMIDA >= limite3) & (cadastro.RENDA_PRESUMIDA < limite2), 'outlier_renda']=0
cadastro.loc[(cadastro.RENDA_PRESUMIDA > limite4) & (cadastro.RENDA_PRESUMIDA < limite3) , 'outlier_renda']=1
cadastro.loc[cadastro.RENDA_PRESUMIDA < limite4 , 'outlier_renda']=2

In [ ]:
# Tabela de frequência da variável outlier_renda

freq_outlier_renda = pd.pivot_table(cadastro,index='outlier_renda', values='NUM_CPF', aggfunc=np.count_nonzero)
freq_outlier_renda['%'] = ((pd.pivot_table(cadastro,index='outlier_renda', values='NUM_CPF', aggfunc=np.count_nonzero)/cadastro['NUM_CPF'].count())*100).round(2)
freq_outlier_renda

In [ ]:
################################################
########### REMOVER OS OUTLIERS ############
################################################

# Removendo os Outliers
cadastroSemOutlier = cadastro.loc[cadastro.outlier_renda == 0.0]

print('Mínimo = ', min(cadastroSemOutlier['RENDA_PRESUMIDA']))

print('Máximo = ', max(cadastroSemOutlier['RENDA_PRESUMIDA']))

print('Média = ', (cadastroSemOutlier['RENDA_PRESUMIDA']).mean())

print('STD = ', cadastroSemOutlier['RENDA_PRESUMIDA'].std())

print('Coeficiente de Variação = ', (cadastroSemOutlier['RENDA_PRESUMIDA'].std() / cadastroSemOutlier['RENDA_PRESUMIDA'].mean()))

print('Mediana = ', (cadastroSemOutlier['RENDA_PRESUMIDA']).median())


In [ ]:
# Gráfico histograma da Ingestão de Sal e Z_score_sal
fig, axes = plt.subplots(nrows=1, ncols=2,figsize=(20,4))

plt.subplot(1,2,1)
plt.hist(cadastro['RENDA_PRESUMIDA'], bins=10)
plt.ylabel('Frequência')
plt.xlabel('RENDA_PRESUMIDA (R$)')
plt.subplot(1,2,2)
plt.hist(cadastroSemOutlier['RENDA_PRESUMIDA'], bins=10)
plt.ylabel('Frequência')
plt.xlabel('RENDA_PRESUMIDA (R$)')

### Exploratory Data Analysis (EAD) - Sweetviz

In [ ]:
!pip install sweetviz

In [ ]:
import sweetviz as sv
import pandas as pd

In [ ]:
# ---- Fix Sweetviz expecting np.VisibleDeprecationWarning (removed in NumPy ≥1.24)
import numpy as np
if not hasattr(np, "VisibleDeprecationWarning"):
    class VisibleDeprecationWarning(Warning):
        pass
    np.VisibleDeprecationWarning = VisibleDeprecationWarning
# ---- Now it's safe to import Sweetviz
import sweetviz as sv
import pandas as pd

In [ ]:
report = sv.analyze(cadastro)
report.show_html("report.html")